### ISSUE TITLE
a question about the loss function in the code

### ISSUE THREAD

**Issue by LJP45654** [yichen928/ActiveFT#2]
in: ActiveFT_CIFAR.py/class SampleModel/def get_loss:
`J = torch.log(prod_exp_pos) - torch.log(prod_exp_pos + cent_prob_exp_sum[pos_k] * self.balance)`
`J = -torch.mean(J)`
It seems that the loss function represented by this code differs from the one mentioned in the article. The return value of the `get_loss` function I wrote based on your code is shown in the following figure,am I mistaken in my understanding about the code?
![image-20230611220122657](https://raw.githubusercontent.com/LJP45654/md-img/main/202306112201802.png)

**Comment by yichen928** [comment]
Thank you for your interest. I have double-checked the code. I think your formula is correct, but it is actually similar to the equation inside our paper. 

The formula in this paper can be re-written as: 
<img width="116" alt="image" src="https://github.com/yichen928/ActiveFT/assets/48796750/8a2de43a-daa6-4167-ba66-663c65b077f5">

<img width="210" alt="image" src="https://github.com/yichen928/ActiveFT/assets/48796750/032650c8-86e5-4ab7-a24b-06b0c9895a88">

This item is same as the paper.

<img width="344" alt="image" src="https://github.com/yichen928/ActiveFT/assets/48796750/8e55350f-c0fe-48b5-9eeb-35f2018edb82">

This is slightly different to our paper, where we actually do not eliminate the similarity of each parameter to itself in the implementation. Since this similarity to itself is always one, it does not affect a lot.

Another difference is that lambda should be set as B/N instead of 1. I will fix this in our next version of paper.
